src/extension/extension.ts

```diff
@@ -622,250 +566,225 @@ export async function activate(ctx: sourcegraph.ExtensionContext): Promise<void>
+            traceAsyncGenerator('Provide implementations', tracer, undefined, async function*(span) {
+                if (canGenerateTraceUrl(span)) {
+                    logger.log('Implementation trace', span.generateTraceURL())
+                }
 
-                    const textDocumentUri = new URL(textDocument.uri)
-                    const serverRootUri = resolveServerRootUri(textDocumentUri, serverSgEndpoint)
-                    const serverTextDocumentUri = toServerTextDocumentUri(textDocumentUri, serverSgEndpoint)
-                    const connection = await getOrCreateConnection(serverRootUri, { span, token })
-                    const implementationParams: TextDocumentPositionParams = {
-                        textDocument: { uri: serverTextDocumentUri.href },
-                        position,
-                    }
-                    const implementationResult = (await sendTracedRequest(
-                        connection,
-                        ImplementationRequest.type,
-                        implementationParams,
-                        { span, tracer, token }
-                    )) as Location[] | Location | null
-                    rewriteUris(implementationResult, toSourcegraphTextDocumentUri)
-                    return convertLocations(implementationResult)
-                })
-            providers.add(
-                sourcegraph.languages.registerLocationProvider(IMPL_ID, documentSelector, {
-                    provideLocations: provideImpls,
-                })
-            )
-            const panelView = sourcegraph.app.createPanelView(IMPL_ID)
-            panelView.title = 'Implementations'
-            panelView.component = { locationProvider: IMPL_ID }
-            panelView.priority = 160
-            providers.add(panelView)
-        }
+                const textDocumentUri = new URL(textDocument.uri)
+                const serverRootUri = resolveServerRootUri(textDocumentUri, serverSgEndpoint)
+                const serverTextDocumentUri = toServerTextDocumentUri(textDocumentUri, serverSgEndpoint)
+                const connection = await getOrCreateConnection(serverRootUri, { span, token })
+                const implementationParams: TextDocumentPositionParams = {
+                    textDocument: { uri: serverTextDocumentUri.href },
+                    position,
+                }
+                const implementationResult = (await sendTracedRequest(
+                    connection,
+                    ImplementationRequest.type,
+                    implementationParams,
+                    { span, tracer, token }
+                )) as Location[] | Location | null
+                rewriteUris(implementationResult, toSourcegraphTextDocumentUri)
+                yield convertLocations(implementationResult)
+            })
+
+        activateCodeIntel(
+            ctx,
+            documentSelector,
+            handlerArgs,
+            config.value['typescript.serverUrl']
```

Comment:
Note that before, you could change the config during a session and it would be considered. Now it seems the config is read only once during activation. I think this could cause problems in particular in the browser extension, where extensions are run in the background page and are not necessarily re-activated just because the page got reloaded.

Reply:
Ah you're right - I missed that. We used to check on each function invocation. I'll update the new API to support doing this dynamically.

Reply:
See https://github.com/sourcegraph/sourcegraph-basic-code-intel/pull/198#issuecomment-579463808.